ls1intum/Orion
# [chore] fix codacy problems, add documentation, add translations

### Description
This PR improves code quality by resolving issues detected by the static code analysis, adding documentation comments and translating more strings.

### Steps for Testing
1.  Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests).
2.  Verify the following changed components work properly:
     - The ChooseImportPath dialog when importing a new exercise should be translated now.
     - The BrokenLinkWarning dialog should now show correctly and be translated. It can be caused by editing the `exerciseId` in `.idea/workspace.xml` (at the bottom), severing the connection between the exercise and the global registry. Warning: This will most likely cause troubles with that repository, so don't use one you need
     - Submitting was slightly refactored but should still work correctly.
     - Reloading from the settings ("Restart Browser") or reloading due to changing the URL or useragent should now correctly show the submit button on the opened exercise, similar to the "return to exercise" button.
3.  [Optional] Review refactored code
